dodo.py

```diff
@@ -81,20 +81,81 @@ def task_openspiel_compile():
     }
 
 
-def task_forecast():
+#####################
+# FORECASTING TASKS #
+#####################
+
+FORECAST_QUERY_LOG_DIR = Path("./forecast/data/extracted/extended/")
+
+FORECAST_ARTIFACTS_DIR = Path("artifacts")
+FORECAST_PREPROCESSOR_ARTIFACT = FORECAST_ARTIFACTS_DIR.joinpath(
+    "preprocessed.parquet.gzip"
+)
+FORECAST_CLUSTER_ARTIFACT = FORECAST_ARTIFACTS_DIR.joinpath("clustered.parquet")
+FORECAST_MODEL_DIR = FORECAST_ARTIFACTS_DIR.joinpath("models")
+FORECAST_PREDICTION_CSV = FORECAST_ARTIFACTS_DIR.joinpath("forecast.csv")
+
+FORECAST_PREDICTION_START = "2021-12-06 14:24:32 EST"
+FORECAST_PREDICTION_END = "2021-12-06 14:24:36 EST"
+
+
+def task_forecast_preprocess():
     """
-    Forecast: forecast the workload.
+    Forecast Preprocess: create templatized queries data for ingest into
+    forecaster training or prediction
     """
+
+    preprocessor_action = (
+        "python3 ./forecast/preprocessor.py"
+        f" --query-log-folder {FORECAST_QUERY_LOG_DIR.absolute()} "
+        f"--output-parquet {FORECAST_PREPROCESSOR_ARTIFACT.absolute()}"
+    )
+
     return {
         "actions": [
-            "mkdir -p artifacts",
+            f"mkdir -p {FORECAST_ARTIFACTS_DIR.absolute()}",
             # Preprocess the PostgreSQL query logs.
-            "python3 ./forecast/preprocessor.py --query-log-folder ./forecast/data/extracted/extended/ --output-hdf ./artifacts/preprocessor.hdf",
-            # Cluster the processed query logs.
-            # TODO(WAN): clusterer shouldn't go directly to the predictions, plug in Mike's part.
-            "python3 ./forecast/clusterer.py --preprocessor-hdf ./artifacts/preprocessor.hdf --output-csv ./artifacts/forecast.csv",
+            preprocessor_action,
+        ],
+        "targets": [FORECAST_PREPROCESSOR_ARTIFACT.absolute()],
```

Comment:
See above comment for absolute(); in general the absolute path should be taken at point of declaration, I think.